chainstory/linkapp/consumers.py

```diff
@@ -1,57 +1,96 @@
-from channels.generic.websocket import AsyncWebsocketConsumer
-from asgiref.sync import sync_to_async
 import json
+from channels.generic.websocket import AsyncWebsocketConsumer
+from channels.db import database_sync_to_async
+from .models import ChatMessage, User
+from django.urls import reverse
+from django.test import Client
 
-class MyConsumer(AsyncWebsocketConsumer):
+class ChatConsumer(AsyncWebsocketConsumer):
     async def connect(self):
-        # Join the chat group
-        await self.channel_layer.group_add("chat", self.channel_name)
+        self.room_group_name = "chat"
+        await self.channel_layer.group_add(self.room_group_name, self.channel_name)
         await self.accept()
-        
-        # Get messages using sync_to_async
-        from .models import ChatMessage  # Import here to avoid circular import
-        get_messages = sync_to_async(lambda: list(ChatMessage.objects.all()))
-        messages = await get_messages()
-        
-        # # Send existing messages
-        # for message in messages:
-        #     await self.send(text_data=json.dumps({
-        #         'message': message.content,
-        #         'username': message.username
-        #     }))
 
     async def disconnect(self, close_code):
-        # Leave the chat group
-        await self.channel_layer.group_discard("chat", self.channel_name)
+        await self.channel_layer.group_discard(self.room_group_name, self.channel_name)
 
     async def receive(self, text_data):
-        from .models import ChatMessage
         text_data_json = json.loads(text_data)
+        
+        if text_data_json.get('type') == 'reset':
+            # Broadcast reset message to all clients
+            await self.channel_layer.group_send(
+                self.room_group_name,
+                {
+                    'type': 'chat_message',
+                    'message': '',
+                    'reset': True
+                }
+            )
+            return
+        
+        # Handle different message types
+        message_type = text_data_json.get('type')
```

Comment:
I would move this up and then use it in your if statement above

Reply:
done